### DISCUSSION TITLE
Present mapsforge maps (06/24) have display issues? - They are somehow bluish - See at the end recommendation for temporary solution

### DISCUSSION THREAD

**Discussion by DIrgi** [mapsforge/mapsforge#1496] [Q&A] [answered]
![image](https://github.com/mapsforge/mapsforge/assets/57968526/43eb8459-03e5-4725-8090-1e63283954c5)
I recently downloaded some mapsforge updates. And I eplored that some have a kind of a blue veil background.
I assume there was an update issue on some maps.

every map except Romania and Hungary are still on May version
Where Romania is complete blue, Hungary it's just a blue parted.
![image](https://github.com/mapsforge/mapsforge/assets/57968526/367fdb96-96c9-492a-a8c6-4eedc87eec94)

Far aways zoom:
![image](https://github.com/mapsforge/mapsforge/assets/57968526/7c4ee8be-e46a-4f5e-b6d1-49bd8074a8ba)

Closer zoom 9.5 Germany:
![image](https://github.com/mapsforge/mapsforge/assets/57968526/efc7d1c1-0bf5-469c-a7c9-8305af855c7d)

Closer zoom 9.5 Romania:
you see this kind of "blue" in between
![image](https://github.com/mapsforge/mapsforge/assets/57968526/31681fe2-dac0-4594-822a-6b42df831574)

On some area it vanishes when zoomin in:
compare zoom: 11.3 vs zoom 12.3 
![image](https://github.com/mapsforge/mapsforge/assets/57968526/c3db4b4a-2b16-4275-a1ac-ea41288b35d9)
![image](https://github.com/mapsforge/mapsforge/assets/57968526/8f14b4c2-c711-4a5f-85be-526c153f5ae3)

On some it remain completly up to max zoom:
![image](https://github.com/mapsforge/mapsforge/assets/57968526/42a9a3a0-47f0-4efb-a8a2-97cc644d0400)

you remember in the first photos, there's even a cut in the hungary map:
![image](https://github.com/mapsforge/mapsforge/assets/57968526/14119f67-8a30-4f14-ab74-070755098fd0)
![image](https://github.com/mapsforge/mapsforge/assets/57968526/32ba47d6-50c1-4e8e-9ca7-a2938f3d1e7d)
![image](https://github.com/mapsforge/mapsforge/assets/57968526/51ea20a3-5f9d-4936-90c3-386e21857426)
![image](https://github.com/mapsforge/mapsforge/assets/57968526/2b654769-cf26-4386-99dc-b67973f6235b)
![image](https://github.com/mapsforge/mapsforge/assets/57968526/5e4bd082-f3fa-4b8a-8fc9-18e4e4e453e1)

zoomed in more closer.

![image](https://github.com/mapsforge/mapsforge/assets/57968526/9744c4a5-e932-4e45-86ff-9430b43edeed)

**Comment by ludwigb** [comment]
This is a known issue, a file used in the update process was corrupted
resulting in the maps compiled between June 18 and June 28 missing the land
layer.

The map files are in the process of being recreated, but it takes time.
Please wait until the map you require has a timestamp after June 28.

Our apologies

Ludwig

DIrgi ***@***.***> schrieb am So. 30. Juni 2024 um 23:52:

**Reply by DIrgi** [reply]
Hello Ludwig, thanks for your feedback. Is there a chance to download the previous intact maps from may? - or do you think this will be solved within one week- cause I need them in two weeks.
I know, technically they‘ll probably work, but it looks awful.

Cheers, Dirk

**Comment by ludwigb** [comment]
I expect this to be fixed by the end of the week.

DIrgi ***@***.***> schrieb am Mo. 1. Juli 2024 um 08:52:

**Reply by DIrgi** [reply]
This sounds great, I can live with that, thank you Ludwig.

**Comment by devemux86** [comment]
There have been announcements in Mapsforge and Cruiser repositories:
- https://github.com/mapsforge/mapsforge/issues/1489

- https://github.com/devemux86/cruiser/discussions/580

Mapsforge maps are already being updated.

**Comment by ludwigb** [comment]
The maps for Romania and Hungary have been recreated and the issue you
reported should be fixed in them.

If there are other maps that you require that exhibit the problem please
let me know

**Reply by DIrgi** [reply]
Hi Ludwig, thank you for the update I'm just checking - yes Hungary and Romania seems to bee fine.
Parts of Czech Republic which are close to Austria still have issues,  as well as complete Austria.
The Alps itself are ok and fortunately cover most of Austria.

But..interesting, I though if Czechia is selected, everything is in up to the border but, what you see here in the picture is CZ and ALPS are on and Austria is OFF.
There's a blue stripe which is still comming from the CZ Map. when I turn it off, it's gone.


![image](https://github.com/mapsforge/mapsforge/assets/57968526/9cf3c1c8-b273-4d9e-b13b-4eacd4cd63d2)

As well have a look on the 2nd Map from Austria (Alps OFF), still partly blue covered. but not all. However it covers as well parts of Italy and Slovakia.
![image](https://github.com/mapsforge/mapsforge/assets/57968526/aea69c4e-b52b-4d6e-9fad-6c5531a975f9)

**Reply by devemux86** [reply]
@DIrgi thanks for the tests!

Maps with date 2024-06-18 to 2024-06-27 are not correct and will be updated, see https://github.com/mapsforge/mapsforge/issues/1489.

**Reply by DIrgi** [reply]
Well I just checked these maps. At, (parts of) De, cz, pl, sk, hu, ro, bg, md, slo, hr, srb, Kosovo.
What’s about other maps I.e. iberia, nordic, greece…- I don’t know.

**Reply by devemux86** [reply]
Also with multiple maps, better test with Mapsforge engine.
(OpenGL is not perfect with overlapping maps)

Can test with [desktop Cruiser](https://github.com/devemux86/cruiser) (download from Releases).

Android **Cruiser 5** will exclusively have the Mapsforge engine.

**Comment by ludwigb** [comment]
I just checked the cz and cyprus map and they still show the problem even
though they have been recreated and should be ok. I am just rebuilding the
cyprus map to see if this is an ongoing issue. The land polygons that were
corrupted in the period reported are ok now on the server.

**Reply by DIrgi** [reply]
I can check in the evening some other maps as well but I’ll do it due to capacity reasons on my desktop PC 😉

**Reply by devemux86** [reply]
@ludwigb 

I do not see any problem with cyprus.map:

![Cruiser](https://github.com/mapsforge/mapsforge/assets/3484020/a34bce67-9531-4df8-b4c3-d651cbe8bca8)

**Comment by ludwigb** [comment]
Unfortunately, the rebuilt cyprus map still does not have the correct land
files, even though the land polygon data seems ok (the previous issue was
caused by that data file being empty).

**Reply by devemux86** [reply]
@ludwigb where exactly?

**Comment by ludwigb** [comment]
Cyprus and Czech Rep are both entirely blue -- just downloaded and viewed
with Cruiser.

No idea what is going on there.

**Reply by devemux86** [reply]
@ludwigb this is strange, are you sure testing the correct cyprus.map and not a cached download?

- cyprus.map (2024-07-07) works correctly in my [tests](https://github.com/mapsforge/mapsforge/discussions/1496#discussioncomment-9978135).

- czech-republic.map does not work, but it has date 2024-06-28 when the rebuilding started.

Probably maps with date 2024-06-28 must be rebuilt, as we are not sure which ones are really new.

**Comment by DIrgi** [comment]
So, here's an update from my side. I downloaded or updated meanwhile **ALL European Maps** incl. **WORLD** , **ALPS** and **CANARIES** to the MAP status existing on 07.07.2024

What I did not download are the submaps i.e. from German, France etc. as well as other Parts of the World incl. Russia

What I can say is (starting with the strange Cyprus topic)
When WORLD is active then it work like a "clean" overlay and Cyprus looks "OK". When I turn "WORLD" off, Cyprus has this blue flooding's.

Cyprus WITH active WORLD Map:
![image](https://github.com/mapsforge/mapsforge/assets/57968526/d5f7c582-29a6-4691-a77d-e1e471467469)

Cyprus with deactivated WORLD Map:
![image](https://github.com/mapsforge/mapsforge/assets/57968526/b8d5e1b5-4c0e-4419-bd11-be7608ed34d5)

![image](https://github.com/mapsforge/mapsforge/assets/57968526/6c2a3abf-b797-46f6-af20-e8cbf105606c)


In general it seems, when "WORLD" Map is activated it cleans out (as over- or underlay (I don't know)) all floodings's on every European map.  When WORLD is OFF, following maps have still blue flooding issues:

- Austria
- Czech Republic
- Bulgaria
- Cyprus
- Danmark
- Belarus

All other European MAPS based under the folder seems to be OK (as mentioned subfolder MAPS of Germany, France, Great-Britain & Italy) not tested.

Hope this helps



-----------------------------------------

Appendix...ups I recognized as well some overlapping influences don't know if this is all over Europe and I don't want to test all map neighborhood combinations. But what I can say is:
If ITALY Map is activated it overlaps somehow Crotia and Bosnia. If Italy is NOT activated and just Croatia and Bosnia, they have damages as well see pictures:

See Bosnia and Croatia (both activated) with ACTIVATED Italy map:
![image](https://github.com/mapsforge/mapsforge/assets/57968526/3e194341-e492-43d7-a667-217643e3bb5b)

See Bosnia and Croatia (both activated) with DEACTIVATED Italy map:
![image](https://github.com/mapsforge/mapsforge/assets/57968526/ed55086f-5bfe-4909-8a2f-19ea865e450d)

btw. in all appedix cases, WORD MAP is deactivated -> this is when WORLD is activated...all is gone again.
![image](https://github.com/mapsforge/mapsforge/assets/57968526/5adf8fa2-538e-4039-adde-ed4130b2436e)


So finally we have two cases:
When WORLD MAP is activated it works as a general cleaner for the MPAS who are still affected.
When WORLD is OFF, some "clean" neighbor countries are working as cleaning filter for their neighbor countries which are affected.

I wonder which wrong "magical" button you guys pressed for the map creations these days 

🤔😂🤪

### 
**Recommendation:**
As longs as some maps still have this issues I would recommend to download the WORD map and activate it then the issues are temporary gone.

**Reply by devemux86** [reply]
@DIrgi thanks for the tests!

The wrong maps don't have land polygons, so if you use the world.map, it helps with that.

All the maps you mentioned have dates between **2024-06-18 to 2024-06-28**,
so they are considered not correct and will be updated anyway.

We should test maps dated **2024-06-29** and beyond.

About cyprus.map (2024-07-07) I cannot confirm it.
I downloaded it again today and it works correctly like seen [above](https://github.com/mapsforge/mapsforge/discussions/1496#discussioncomment-9978135).

**Reply by DIrgi** [reply]
Well regarding Cyprus, I can only tell you what I see and what happened. I mean you see my shot as well.

„..the wrong maps don‘t have land polygons…“ and what about Austria and CZ? They have no coast except lakes but this has every country.

**Reply by devemux86** [reply]
All maps have land polygons with or without coastline, this is also needed for lakes, etc.

**Comment by applantation** [comment]
There seems to be some weird caching going on.

I copied the cyprus map that I regenerated yesterday to
https://download.mapsforge.org/maps/v5/europe/cyprustest.map. When I
download that map I do not get the error and the map is fine, as it should
be.

But when I copied the same map yesterday morning, I downloaded the old map
(their sizes are different). When I now download the cyprus.map (not the
copy) I also get the good copy.

I do not know where the caching comes from, the files are just served by
apache on my server and I have not set up any caching on my server for such
files.

Dirgi -- could you check what happens if you use the cyprustest.map file?

Thanks

**Reply by DIrgi** [reply]
I‘ll do so that evening.

**Reply by devemux86** [reply]
@DIrgi 

@ludwigb means to download directly from the Mapsforge server:
https://download.mapsforge.org/maps/v5/europe/

You can use Android or desktop Cruiser for the tests and open it as external map (enable the "Extended menu").

**Reply by DIrgi** [reply]
I do my tests first on a virtual android surface (based on Android 9) on a Desktop PC. when it works there I do it on my android tablet which is android 11. 
I need my tablet stable and safe since I go on vacation that friday. For that reason I updated & downloaded the (for me rather not common) maps from HU and RO . This was the reason why I saw the issue. If I would have stayed with May maps, I would not even be aware about.
Thanks devemux, I‘m aware about the extended menu. Theses are always the buttons I press first in a new software 😉

**Reply by DIrgi** [reply]
Btw I sometimes hear about the desktop cruiser version. Where I can find it. Does it work on a Win PC? Or is it Linux based?

**Reply by devemux86** [reply]
@DIrgi 

Desktop Cruiser is a Java app so it works on all desktop platforms: Windows, macOS, Linux, etc.

You can download the zip from the [Cruiser website](https://github.com/devemux86/cruiser) under "Releases" on the right.

**Reply by DIrgi** [reply]
**Off Topic:** Regarding the cruiser desktop Java version.
I did download and open it, yes it works but well compare to Android version it has some limitations when it comes to the options.
I think I'll remain with my virtual Android surface it's very close to the Android version on an Android hardware.
If you are interested and want to check it out, you can get the  virtual OS for a windows PC here:
https://en.ldplayer.net/?from=en

The main purpose of the Android emulator is for people like to play Android games on a PC but no worries it's basically a Android 9 OS and you can pretty well install Cruiser and act like being on and Android smartphone. Check it out if you like.

Some screen shots:
![image](https://github.com/mapsforge/mapsforge/assets/57968526/ee53375e-7e6f-4d07-8689-ce53141651a7)

![image](https://github.com/mapsforge/mapsforge/assets/57968526/53846901-4139-490f-b9a6-e3a64b0c0ff9)

![image](https://github.com/mapsforge/mapsforge/assets/57968526/62b3bec0-f020-467d-b187-6729eabadece)

![image](https://github.com/mapsforge/mapsforge/assets/57968526/bee80089-c246-4140-96e9-a6c6d8e966d8)

**Reply by devemux86** [reply]
The current differences are:
- Desktop Cruiser uses the Mapsforge map engine
- Android Cruiser 4 uses the VTM (OpenGL) map engine
- Android Cruiser 5 will use the Mapsforge map engine

Also Android offers more options than the desktop version.

**Comment by ludwigb** [comment]
Thanks

**Reply by DIrgi** [reply]
Hi Ludwig, I downloaded the test map and opened it into Cruiser:
As I can see, this test map is the only active one.
This is how it looks like.
Everything seems to be fine
<img width="533" alt="image" src="https://github.com/mapsforge/mapsforge/assets/57968526/c320c848-d6cc-43da-8e8e-d4763da5d5bf">

**Comment by devemux86** [comment, marked as answer]
Mapsforge maps with the problem were all updated.

If you see anything else, please let us know.

**Reply by DIrgi** [reply]
First check of:
- Austria
- Czech Republic
- Bulgaria
- Cyprus
- Belarus

looks ok,

- Danmark

I haven't checked.

**Comment by ludwigb** [comment]
Hi Dirgi, thanks for checking, Denmark should also be ok, because the
underlying cause has been fixed and the map rebuilt.

Best regards

Ludwig

DIrgi ***@***.***> schrieb am Do. 11. Juli 2024 um 20:49:
